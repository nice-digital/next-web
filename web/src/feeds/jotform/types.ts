export type FormID = `${number}`;

export type Duration = `${number}ms`;

export type NotFoundResponse = {
	responseCode: 404;
	message: string;
	content: "";
	duration: Duration;
	info: string;
};

export type NotAuthorizedResponse = {
	responseCode: 401;
	message: string;
	content: "";
	duration: Duration;
	info: string;
};

export type FormStatus = "ENABLED" | "DISABLED";

export type FormType = "LEGACY" | "CARD";

export type FormSuccessResponse = {
	responseCode: 200;
	message: "success";
	content: {
		id: string;
		username: string;
		title: string;
		height: number;
		status: FormStatus;
		/** Format like "2023-03-21 09:27:18" */
		created_at: string;
		updated_at: string;
		last_submission: string;
		new: `${number}`;
		count: `${number}`;
		type: FormType;
		favorite: `${number}`;
		archived: `${number}`;
		url: string;
	};
	duration: Duration;
};

export type GetFormResponse =
	| NotAuthorizedResponse
	| NotFoundResponse
	| FormSuccessResponse;
