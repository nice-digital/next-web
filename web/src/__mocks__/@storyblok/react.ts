export const storyblokInit = (): null => null;

export const apiPlugin = (): null => null;
