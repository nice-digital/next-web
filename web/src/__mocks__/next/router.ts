export const useRouter = jest.fn().mockImplementation(() => {
	throw new Error("Provide a mock of useRouter for testing purposes");
});
