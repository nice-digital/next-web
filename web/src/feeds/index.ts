import { type Readable } from "stream";

import axios, { type AxiosInstance } from "axios";
import applyCaseMiddleware from "axios-case-converter";
import { camelCase } from "camel-case";

import { cache, getCacheKey } from "@/cache";

export const client: AxiosInstance = applyCaseMiddleware(
	axios.create({
		headers: { accept: "application/json" },
	}),
	{
		caseFunctions: {
			camel: (input, _options) => {
				// Strip out the prefix from all the embedded keys as it creates so much noise
				return camelCase(input.replace("nice.publications:", ""));
			},
		},
	}
);

/**
 * Gets the body of a feed directly from the back end system
 *
 * @param path The path of the feed to request
 * @returns The body of the feed
 */
export const getFeedBodyUnCached = async <TResponse>(
	origin: string,
	path: string,
	apiKey: string,
	acceptHeader = "application/json"
): Promise<TResponse> => {
	const { data } = await client.get<TResponse>(origin + path, {
		headers: {
			"Api-Key": apiKey,
			Accept: acceptHeader,
		},
		validateStatus: (status: number) => {
			// We don't want feed 404 responses to throw an error, so that we can show users a not found page rather than a server error.
			return (status >= 200 && status < 300) || status == 404;
		},
	});

	return data;
};

/**
 * Gets the body of a feed from the cache, if it exists.
 * If the item isn't in cache then it will call the `getUncachedAction` function to get fresh data.
 *
 * @param path The path of the feed endpoint to request
 * @param ttl The TTL (time to live) of the entry
 * @param getUncachedAction Function that gets the uncached feed directly from the back end.
 * @returns The body of the feed
 */
export const getFeedBodyCached = async <T>(
	groupCacheKey: string,
	path: string,
	ttl: number,
	getUncachedAction: () => Promise<T>
): Promise<T> =>
	cache.wrap<T>(getCacheKey(groupCacheKey.toLowerCase(), path.toLowerCase()), getUncachedAction, { ttl });

/**
 * Gets a response stream from a remote API endpoint, usually used for binary files e.g. PDFs (or mobi/epub etc).
 *
 * @param origin The base URL of the endpoint
 * @param path The path to the endpoint e.g. `/feeds/downloads/737585a0-dad7-4a37-875b-b30b09c3fdc3`
 * @param apiKey The API key for authentication against the API
 * @returns A readable response stream
 */
export const getResponseStream = async (
	origin: string,
	path: string,
	apiKey: string
): Promise<Readable> => {
	const { data } = await axios.get<Readable>(origin + path, {
		headers: {
			"Api-Key": apiKey,
		},
		responseType: "stream",
		maxBodyLength: Infinity,
	});
	return data;
};
