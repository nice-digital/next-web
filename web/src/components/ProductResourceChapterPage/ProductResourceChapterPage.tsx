import { NextSeo } from "next-seo";
import { FC } from "react";

import { Breadcrumbs, Breadcrumb } from "@nice-digital/nds-breadcrumbs";
import { Grid, GridItem } from "@nice-digital/nds-grid";

import { FileSize } from "@/components/FileSize/FileSize";
import { Link } from "@/components/Link/Link";
import { OnThisPage } from "@/components/OnThisPage/OnThisPage";
import { ProductHorizontalNav } from "@/components/ProductHorizontalNav/ProductHorizontalNav";
import { ProductPageHeading } from "@/components/ProductPageHeading/ProductPageHeading";
import { PublicationsChapterMenu } from "@/components/PublicationsChapterMenu/PublicationsChapterMenu";
import { PublicationsPrevNext } from "@/components/PublicationsPrevNext/PublicationsPrevNext";
import { formatDateStr, stripTime } from "@/utils/datetime";

import { type ProductResourceChapterPageProps } from "./ProductResourceChapterPage.getServerSideProps";
import styles from "./ProductResourceChapterPage.module.scss";

export { type ProductResourceChapterPageProps } from "./ProductResourceChapterPage.getServerSideProps";

export const ProductResourceChapterPage: FC<
	ProductResourceChapterPageProps
> = ({
	productPath,
	product,
	hasToolsAndResources,
	hasInfoForPublicResources,
	hasEvidenceResources,
	hasHistory,
	chapters,
	chapterSections,
	htmlBody,
	title,
	lastUpdated,
	resourceTypeSlug,
	resourceTypeName,
	resourceDownloadPath,
	resourceDownloadSizeBytes,
}) => {
	const hasOnThisPageMenu = chapterSections.length > 1;

	return (
		<>
			<NextSeo
				title={`${title} | ${resourceTypeName} | ${product.id} | Indicators | Standards and Indicators`}
			/>

			<Breadcrumbs>
				<Breadcrumb to="/">Home</Breadcrumb>
				<Breadcrumb to="/standards-and-indicators">
					Standards and Indicators
				</Breadcrumb>
				<Breadcrumb to="/standards-and-indicators/indicators">
					Indicators
				</Breadcrumb>
				<Breadcrumb to={productPath} elementType={Link}>
					{product.id}
				</Breadcrumb>
				<Breadcrumb
					to={productPath + "/" + resourceTypeSlug}
					elementType={Link}
				>
					{resourceTypeName}
				</Breadcrumb>
				<Breadcrumb>{title}</Breadcrumb>
			</Breadcrumbs>

			<ProductPageHeading product={product} />

			<ProductHorizontalNav
				productTypeName="Indicator"
				productPath={productPath}
				hasToolsAndResources={hasToolsAndResources}
				hasInfoForPublicResources={hasInfoForPublicResources}
				hasEvidenceResources={hasEvidenceResources}
				hasHistory={hasHistory}
			/>

			<Grid gutter="loose">
				<GridItem
					cols={12}
					md={4}
					lg={3}
					elementType="section"
					aria-label="Chapters"
				>
					<PublicationsChapterMenu
						ariaLabel="Chapter pages"
						chapters={chapters}
					/>
				</GridItem>

				<GridItem cols={12} md={8} lg={9} elementType="section">
					<Grid reverse gutter="loose">
						{hasOnThisPageMenu ? (
							<GridItem cols={12} md={4} lg={3}>
								<OnThisPage sections={chapterSections} />
							</GridItem>
						) : null}
						<GridItem
							cols={12}
							md={hasOnThisPageMenu ? 8 : 12}
							lg={hasOnThisPageMenu ? 9 : 12}
						>
							<h2 className={styles.heading}>{title}</h2>
							{resourceDownloadPath ? (
								<p>
									<a href={resourceDownloadPath}>
										Download (PDF
										{resourceDownloadSizeBytes ? (
											<>
												, <FileSize fileSizeBytes={resourceDownloadSizeBytes} />
											</>
										) : null}
										)
									</a>
								</p>
							) : null}
							<div dangerouslySetInnerHTML={{ __html: htmlBody }} />
							{lastUpdated ? (
								<p>
									This page was last updated on{" "}
									<time dateTime={stripTime(lastUpdated)}>
										{formatDateStr(lastUpdated)}
									</time>
								</p>
							) : null}
							<PublicationsPrevNext chapters={chapters} />
						</GridItem>
					</Grid>
				</GridItem>
			</Grid>
		</>
	);
};
