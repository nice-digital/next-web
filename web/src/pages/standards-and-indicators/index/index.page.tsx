import { GetServerSideProps } from "next/types";

export const getServerSideProps: GetServerSideProps = async () => {
	return {
		redirect: {
			destination: "/indicators/published",
			permanent: true,
		},
	};
};

export default function OldIndicatorListPage(): void {
	// Default export to prevent next.js errors: we don't need an actual component
	// because the work is done in getServerSideProps to redirect
}
