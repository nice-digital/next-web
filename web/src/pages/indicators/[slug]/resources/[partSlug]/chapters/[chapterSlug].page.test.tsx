import { render } from "@testing-library/react";
import MockAdapter from "axios-mock-adapter";
import { GetServerSidePropsContext } from "next";
import { useRouter } from "next/router";

import { client } from "@/feeds/index";
import { FeedPath } from "@/feeds/publications/types";
import mockProduct from "@/mockData/publications/feeds/products/pmg20.json";
import mockProductTypes from "@/mockData/publications/feeds/producttypes.json";
import mockResource from "@/mockData/publications/feeds/resource/16615.json";

import ProductResourceChapterPage, {
	getServerSideProps,
} from "./[chapterSlug].page";

const axiosJSONMock = new MockAdapter(client, {
	onNoMatch: "throwException",
});

const productRoot = "process",
	slug = "pmg20",
	resourceId = "1967369",
	partId = "2549710189",
	partSlug = `developing-nice-guidelines-the-manual-appendices-${resourceId}-${partId}`,
	chapterSlug =
		"appendix-b-approaches-to-additional-consultation-and-commissioned-primary-research",
	resolvedUrl = `/${productRoot}/${slug}/${partSlug}/chapters/${chapterSlug}`,
	getServerSidePropsContext = {
		params: {
			slug,
			partSlug,
			chapterSlug,
		},
		resolvedUrl,
		query: {
			productRoot,
		},
	} as unknown as GetServerSidePropsContext<{
		slug: string;
		partSlug: string;
		chapterSlug: string;
	}>;

describe("/product/[slug]/resources/[partSlug]/chapters/[chapterSlug].page", () => {
	beforeEach(() => {
		axiosJSONMock.reset();

		axiosJSONMock
			.onGet(new RegExp(FeedPath.ProductDetail))
			.reply(200, mockProduct)
			.onGet(new RegExp(FeedPath.ProductTypes))
			.reply(200, mockProductTypes)
			.onGet(new RegExp(FeedPath.ProductTypes))
			.reply(200, mockProductTypes)
			.onGet(new RegExp("/feeds/resource/16615"))
			.reply(200, mockResource);

		(useRouter as jest.Mock).mockReturnValue({
			asPath: resolvedUrl,
		});
	});

	describe("ProductResourceChapterPage", () => {
		it("should match snapshot", async () => {
			const result = await getServerSideProps(getServerSidePropsContext);

			if ("notFound" in result || "redirect" in result) {
				throw Error(`Expected props not notFound or redirect`);
			}

			const props = await Promise.resolve(result.props),
				{ container } = render(<ProductResourceChapterPage {...props} />);

			expect(container).toMatchSnapshot();
		});
	});

	describe("getServerSideProps", () => {
		it("should return props for valid product, resource and part", async () => {
			expect(
				await getServerSideProps(getServerSidePropsContext)
			).toMatchSnapshot();
		});
	});
});
