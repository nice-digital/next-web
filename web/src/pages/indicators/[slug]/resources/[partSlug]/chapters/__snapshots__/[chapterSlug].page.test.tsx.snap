// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/product/[slug]/resources/[partSlug]/chapters/[chapterSlug].page ProductResourceChapterPage should match snapshot 1`] = `
<div>
  <nav
    aria-label="Breadcrumbs"
    role="navigation"
  >
    <ol
      class="breadcrumbs"
    >
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/standards-and-indicators"
        >
          Standards and Indicators
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/standards-and-indicators/indicators"
        >
          Indicators
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/process/pmg20"
        >
          PMG20
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <a
          href="/process/pmg20/resources"
        >
          Tools and resources
        </a>
      </li>
      <li
        class="breadcrumbs__crumb"
      >
        <span>
          Developing NICE guidelines: the manual – appendices
        </span>
      </li>
    </ol>
  </nav>
  <script
    type="application/ld+json"
  >
    {
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [
    {
      "@type": "ListItem",
      "position": 1,
      "item": {
        "@id": "/",
        "name": "Home"
      }
    },
    {
      "@type": "ListItem",
      "position": 2,
      "item": {
        "@id": "/standards-and-indicators",
        "name": "Standards and Indicators"
      }
    },
    {
      "@type": "ListItem",
      "position": 3,
      "item": {
        "@id": "/standards-and-indicators/indicators",
        "name": "Indicators"
      }
    },
    {
      "@type": "ListItem",
      "position": 4,
      "item": {
        "@id": "/process/pmg20",
        "name": "PMG20"
      }
    },
    {
      "@type": "ListItem",
      "position": 5,
      "item": {
        "@id": "/process/pmg20/resources",
        "name": "Tools and resources"
      }
    },
    {
      "@type": "ListItem",
      "position": 6,
      "item": {
        "name": "Developing NICE guidelines: the manual – appendices"
      }
    }
  ]
}
  </script>
  <div
    class="page-header"
    id="content-start"
  >
    <h1
      class="page-header__heading page-header__heading--alt"
    >
      Developing NICE guidelines: the manual 
    </h1>
    <ul
      class="page-header__metadata"
    >
      <li>
        Process and methods
      </li>
      <li>
        PMG20
      </li>
      <li>
        Published:
        <time
          datetime="2014-10-31"
        >
           
          31 October 2014
        </time>
      </li>
      <li>
        Last updated:
        <time
          datetime="2022-01-18"
        >
           
           
          18 January 2022
        </time>
      </li>
    </ul>
  </div>
  <nav
    class="horizontal-nav"
  >
    <ul
      class="horizontal-nav__list"
    >
      <li
        class="horizontal-nav__item"
      >
        <a
          aria-current="true"
          class="horizontal-nav__link"
          href="/process/pmg20"
        >
          Indicator
        </a>
      </li>
      <li
        class="horizontal-nav__item"
      >
        <a
          aria-current="false"
          class="horizontal-nav__link"
          href="/process/pmg20/resources"
        >
          Tools and resources
        </a>
      </li>
    </ul>
  </nav>
  <div
    class="grid grid--loose"
  >
    <section
      aria-label="Chapters"
      data-g="12 md:4 lg:3"
    >
      <nav
        aria-label="Chapter pages"
        class="stacked-nav"
      >
        <ul
          class="stacked-nav__list"
        >
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="false"
              href="/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  Appendix A: Service delivery – developing review questions, evidence reviews and synthesis
                </span>
              </span>
            </a>
          </li>
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="false"
              href="/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-b-approaches-to-additional-consultation-and-commissioned-primary-research"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  Appendix B: Approaches to additional consultation and commissioned primary research
                </span>
              </span>
            </a>
          </li>
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="false"
              href="/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-c-key-roles-and-responsibilities-of-committee-members"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  Appendix C: Key roles and responsibilities of committee members
                </span>
              </span>
            </a>
          </li>
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="false"
              href="/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-d-guideline-committee-terms-of-reference-and-standing-orders"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  Appendix D: Guideline committee Terms of Reference and Standing Orders
                </span>
              </span>
            </a>
          </li>
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="false"
              href="/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-e-code-of-conduct-for-committee-members"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  Appendix E: Code of conduct for committee members
                </span>
              </span>
            </a>
          </li>
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="false"
              href="/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-f-suggested-sources-for-scoping"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  Appendix F: Suggested sources for scoping
                </span>
              </span>
            </a>
          </li>
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="false"
              href="/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-g-sources-for-evidence-reviews"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  Appendix G: Sources for evidence reviews
                </span>
              </span>
            </a>
          </li>
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="false"
              href="/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-h-appraisal-checklists-evidence-tables-grade-and-economic-profiles"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  Appendix H: Appraisal checklists, evidence tables, GRADE and economic profiles
                </span>
              </span>
            </a>
          </li>
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="false"
              href="/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-i-review-protocol-for-add-key-area-for-example-unplanned-hospital-admission-flu"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  Appendix I: Review protocol for [add key area, for example, unplanned hospital admission / Flu vaccination]
                </span>
              </span>
            </a>
          </li>
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="false"
              href="/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-j-call-for-evidence-and-expert-witnesses"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  Appendix J: Call for evidence and expert witnesses
                </span>
              </span>
            </a>
          </li>
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="false"
              href="/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-k-network-meta-analysis-reporting-standards"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  Appendix K: Network meta-analysis reporting standards
                </span>
              </span>
            </a>
          </li>
          <li
            class="stacked-nav__list-item"
          >
            <a
              aria-current="false"
              href="/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-l-interim-process-and-methods-for-guidelines-developed-in-response-to-health-and-social"
            >
              <span
                class="stacked-nav__content-wrapper"
              >
                <span>
                  Appendix L: Interim process and methods for guidelines developed in response to health and social care emergencies
                </span>
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </section>
    <section
      data-g="12 md:8 lg:9"
    >
      <div
        class="grid grid--rev grid--loose"
      >
        <div
          data-g="12 md:12 lg:12"
        >
          <h2
            class="heading"
          >
            Developing NICE guidelines: the manual – appendices
          </h2>
          <p>
            <a
              href="/process/pmg20/downloads/PMG20-developing-nice-guidelines-the-manual-appendices-1967369-2549710189.pdf"
            >
              Download (PDF
              , 
              395 kB
              )
            </a>
          </p>
          <div />
          <p>
            This page was last updated on
             
            <time
              datetime="2022-01-18"
            >
              18 January 2022
            </time>
          </p>
          <div
            class="prev-next"
          >
            <nav
              aria-label="Previous and next pages"
            >
              <a
                class="prev-next__link"
                href="/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189"
              >
                <span
                  class="prev-next__link-intro"
                >
                  Next page
                </span>
                <span
                  class="prev-next__link-text"
                >
                  Appendix A: Service delivery – developing review questions, evidence reviews and synthesis
                </span>
              </a>
            </nav>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;

exports[`/product/[slug]/resources/[partSlug]/chapters/[chapterSlug].page getServerSideProps should return props for valid product, resource and part 1`] = `
Object {
  "props": Object {
    "chapterSections": Array [],
    "chapters": Array [
      Object {
        "title": "Appendix A: Service delivery – developing review questions, evidence reviews and synthesis",
        "url": "/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189",
      },
      Object {
        "title": "Appendix B: Approaches to additional consultation and commissioned primary research",
        "url": "/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-b-approaches-to-additional-consultation-and-commissioned-primary-research",
      },
      Object {
        "title": "Appendix C: Key roles and responsibilities of committee members",
        "url": "/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-c-key-roles-and-responsibilities-of-committee-members",
      },
      Object {
        "title": "Appendix D: Guideline committee Terms of Reference and Standing Orders",
        "url": "/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-d-guideline-committee-terms-of-reference-and-standing-orders",
      },
      Object {
        "title": "Appendix E: Code of conduct for committee members",
        "url": "/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-e-code-of-conduct-for-committee-members",
      },
      Object {
        "title": "Appendix F: Suggested sources for scoping",
        "url": "/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-f-suggested-sources-for-scoping",
      },
      Object {
        "title": "Appendix G: Sources for evidence reviews",
        "url": "/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-g-sources-for-evidence-reviews",
      },
      Object {
        "title": "Appendix H: Appraisal checklists, evidence tables, GRADE and economic profiles",
        "url": "/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-h-appraisal-checklists-evidence-tables-grade-and-economic-profiles",
      },
      Object {
        "title": "Appendix I: Review protocol for [add key area, for example, unplanned hospital admission / Flu vaccination]",
        "url": "/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-i-review-protocol-for-add-key-area-for-example-unplanned-hospital-admission-flu",
      },
      Object {
        "title": "Appendix J: Call for evidence and expert witnesses",
        "url": "/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-j-call-for-evidence-and-expert-witnesses",
      },
      Object {
        "title": "Appendix K: Network meta-analysis reporting standards",
        "url": "/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-k-network-meta-analysis-reporting-standards",
      },
      Object {
        "title": "Appendix L: Interim process and methods for guidelines developed in response to health and social care emergencies",
        "url": "/process/pmg20/resources/developing-nice-guidelines-the-manual-appendices-1967369-2549710189/chapters/appendix-l-interim-process-and-methods-for-guidelines-developed-in-response-to-health-and-social",
      },
    ],
    "hasEvidenceResources": false,
    "hasHistory": false,
    "hasInfoForPublicResources": false,
    "hasToolsAndResources": true,
    "htmlBody": undefined,
    "lastUpdated": "2022-01-18T00:00:00",
    "product": Object {
      "id": "PMG20",
      "lastMajorModificationDate": "2022-01-18T00:00:00",
      "productTypeName": "Process and methods",
      "publishedDate": "2014-10-31T00:00:00",
      "title": "Developing NICE guidelines: the manual ",
    },
    "productPath": "/process/pmg20",
    "resourceDownloadPath": "/process/pmg20/downloads/PMG20-developing-nice-guidelines-the-manual-appendices-1967369-2549710189.pdf",
    "resourceDownloadSizeBytes": 394577,
    "resourceTypeName": "Tools and resources",
    "resourceTypeSlug": "resources",
    "title": "Developing NICE guidelines: the manual – appendices",
  },
}
`;
